### DISCUSSION TITLE
Kicking users to login screen based on state?

### DISCUSSION THREAD

**Discussion by WalrusSoup** [infinitered/ignite#1805] [Q&A] [answered]
I've been spending a lot of time thinking on how to handle login \ authentication for my app and was curious how everyone is handling these scenarios:

* If I require login to use my app, and initialize the store, then I need to also re-validate that the user still has a valid access token when launching the app. If they don't, I need to kick them to the login screen.
* If a user is kicked off the platform (banned, etc), then I need to boot them out
* This can happen on basically any route

Given this can happen from almost anywhere, how do you all handle it? Do you wait for them to relaunch the app and just keep returning access denied on the API layer? Is there a way for the state to invoke a dump-all function and take them back to the main screen?

My worry is that in the latter option, I would need each store that calls the API to be aware of the bad API response which seems really cumbersome. It may be better just to show a login screen on app boot and restore state or attempt to validate and go from there?

How are you handling it in your apps?

**Comment by jamonholmgren** [comment, marked as answer]
Hey @WalrusSoup! Good question.

From a user experience standpoint, it makes sense to show a screen (modal or otherwise) that indicates they've been logged out and need to log in again. Something like how Gmail will show you an intermediate screen when you've been logged out.

We often have an API service object (see this: https://github.com/infinitered/ignite/blob/ef5810aac792794ee9102e4e563862145d098ea0/boilerplate/app/services/api/api.ts#L55-L58) which allows you to handle errors at that boundary. However, I'm not particularly happy with how that is set up, and want to find a cleaner solution.
